app/src/main/java/com/bnyro/contacts/ui/components/prefs/BlockPreference.kt

```diff
@@ -36,8 +37,23 @@ fun BlockPreference(
                 .fillMaxWidth()
         ) {
             val cornerRadius = 20.dp
+
             var selectedItem by remember {
-                mutableStateOf(Preferences.getInt(preferenceKey, 0))
+                mutableStateOf(
+                    value = Preferences.getInt(preferenceKey, 0),
+                    policy = object :
+                        SnapshotMutationPolicy<Int> {
+                        override fun equivalent(a: Int, b: Int): Boolean {
+                            val areEquals = a == b
+                            if (!areEquals) {
+                                Preferences.edit {
+                                    putInt(preferenceKey, b)
+                                }
```

Comment:
Why don't we just do that in the `onClick` listener below?

Reply:
I have plans to move this code into its own function and use it everywhere.
This feels like a more cleaner solution.